implementation/building.tex

```diff
@@ -14,9 +14,55 @@ \subsection{Early days}
 \texttt{armboot} is a small template project for using Rust bare metal on a STM32 ARM \glspl{mcu}.
 These are, similarly with the EFM32 series, also based on the Cortex-M series of ARM processor cores.
 
-\todo{Håvard: Skriv denne seksjonen}
 The project got up and running pretty fast based on how \texttt{armboot} was built.
 We looked at armboot's Makefile to figure out what flags to pass to \texttt{rustc} in order to make it cross-compile for an ARM target architecture.
+The components listed in \autoref{tab:build:components} were identified as the minimal set of needed to build an executable for the Gecko.
+
+\begin{table}[H]
+  \centering
+  \begin{tabular}{l l l}
+    thumbv7m-none-eabi.json & llvm spec for compiler backend \\ \hline
+    zero.rs & minimal runtime requirements \\ \hline
+    blinky.rs & a program to execute \\ \hline
+    efm32gg.ld & linker script \\ \hline
+    startup\_efm32gg.s & ResetHandler and interrupt vector \\ \hline
+    system\_efm32gg.c & System Clock management \\
+  \end{tabular}
+  \caption{Initial Components for building Rust for ARM Cortex-M}
+  \label{tab:build:components}
+\end{table}
+
+The components in \autoref{tab:build:components} are the files included in the initial successfull build for the Gecko located at geckoboot.rs \cite{web:geckobook-initial} for references.
+In order to compile these components into an executable a Makefile was developed.
+The compilation process consisted of compiling the \texttt{blinky.rs} file to assembly by passing the target architecture and assembly output flag to the \texttt{rustc} compiler.
+This file was then, along with the startup and system files compiled into object files with \texttt{arm-none-eabi-gcc} and linked into an executable with the linker script using \texttt{arm-none-eabi-ld}.
+
+Note: The initial build did not contain the \gls{RCL}.
```

Comment:
Hva med å inkludere denne setningen inn i forrige avsnitt, istedet for å ha et avsnitt med en setning?

f.eks.
> ...and linked into an executable with the linker script using \texttt{arm-none-eabi-ld}.
Note that the initial build did not contain the \gls{RCL}.